File: eCakeShop.Models/Novost.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCakeShop.Models
{
    public class Novost
    {
        [Browsable(false)]
        public int NovostID { get; set; }
        [DisplayName("Dodao korisnik")]
        public string UposlenikIme => $"{Korisnik?.Ime} {Korisnik?.Prezime}";
        public string Naslov { get; set; }
        public string Sadrzaj { get; set; }
        [DisplayName("Slika")]
        public byte[] Thumbnail { get; set; }
        [DisplayName("Datum kreiranja")]
        public DateTime DatumKreiranja { get; set; }
        [Browsable(false)]
        public int KorisnikID { get; set; }
        [Browsable(false)]
        public Korisnik Korisnik { get; set; }
    }
}

File: UI/e_cakeshop_admin/lib/models/novost.dart
import 'package:e_cakeshop_admin/models/korisnik.dart';
import 'package:json_annotation/json_annotation.dart';
part 'novost.g.dart';

@JsonSerializable()
class Novost {
  int? novostID;
  String? naslov;
  String? sadrzaj;
  String? thumbnail;
  String? opis;
  DateTime? datumKreiranja;
  int? korisnikID;
  Korisnik? korisnik;

  Novost({
    this.novostID,
    this.naslov,
    this.sadrzaj,
    this.thumbnail,
    this.opis,
    this.datumKreiranja,
    this.korisnikID,
    this.korisnik,
  });

  factory Novost.fromJson(Map<String, dynamic> json) => _$NovostFromJson(json);

  Map<String, dynamic> toJson() => _$NovostToJson(this);
}
